### PR TITLE
CEDS-698

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

File: src/test/scala/uk/gov/hmrc/wco/dec/inventorylinking/movement/response/InventoryLinkingMovementResponseFromXmlSpec.scala [hunk 1]
```diff
@@ -14,72 +14,72 @@
  * limitations under the License.
  */
 
-package uk.gov.hmrc.wco.dec.inventoryLinking.movement.response
+package uk.gov.hmrc.wco.dec.inventorylinking.movement.response
```
Comment (PatrykRudnicki) [inline src/test/scala/uk/gov/hmrc/wco/dec/inventorylinking/movement/response/InventoryLinkingMovementResponseFromXmlSpec.scala:17]: Why this change? In my opinion, it should be still `inventoryLinking`

File: src/main/scala/uk/gov/hmrc/wco/dec/inventorylinking/consolidation/request/InventoryLinkingConsolidationRequest.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import uk.gov.hmrc.wco.dec.utilities.JacksonMapper
+
+import scala.collection.JavaConverters._
+
+object InventoryLinkingConsolidationRequest extends JacksonMapper {
+  final val inventoryLinking = "http://gov.uk/customs/inventoryLinking/v1"
+
+  def fromXml(xml: String): InventoryLinkingConsolidationRequest = _xml.readValue(xml, classOf[InventoryLinkingConsolidationRequest])
+
+  def fromProperties(props: Map[String, String]): MetaData = {
```
Comment (raghu1978) [inline src/main/scala/uk/gov/hmrc/wco/dec/inventorylinking/consolidation/request/InventoryLinkingConsolidationRequest.scala]: whats the need for MetaData?
```suggestion
  def fromProperties(props: Map[String, String]): MetaData = {
```

File: src/main/scala/uk/gov/hmrc/wco/dec/Declaration.scala [hunk 3]
```diff
@@ -51,24 +47,6 @@ private[wco] object NS {
   final val rs = "urn:wco:datamodel:WCO:Response_DS:DMS:2"
 }
 
-trait JacksonMapper {
```
Comment (raghu1978) [inline src/main/scala/uk/gov/hmrc/wco/dec/Declaration.scala:54]: can you check that there are no updates for declaration

**Review by PatrykRudnicki — APPROVED** [review]
Generally okay. I don't know why you changed name of package, it should contain this capital letter.

Moreover, if you can please add `TODO` to those methods that we want to change to generic.

File: src/main/scala/uk/gov/hmrc/wco/dec/inventorylinking/consolidation/request/InventoryLinkingConsolidationRequest.scala (hunk 2 above)
Comment (MaciejRewera) [reply src/main/scala/uk/gov/hmrc/wco/dec/inventorylinking/consolidation/request/InventoryLinkingConsolidationRequest.scala]: Missed it during copy-pasting. Thanks!

File: src/test/scala/uk/gov/hmrc/wco/dec/inventorylinking/movement/response/InventoryLinkingMovementResponseFromXmlSpec.scala (hunk 1 above)
Comment (MaciejRewera) [reply src/test/scala/uk/gov/hmrc/wco/dec/inventorylinking/movement/response/InventoryLinkingMovementResponseFromXmlSpec.scala:17]: I guess it depends on naming convention. But yes, I did change the package name, or we would have both 'inventorylinking' and 'inventoryLinking'.

File: src/main/scala/uk/gov/hmrc/wco/dec/Declaration.scala (hunk 3 above)
Comment (MaciejRewera) [reply src/main/scala/uk/gov/hmrc/wco/dec/Declaration.scala:54]: These changes are due to extraction of common Traits and Objects.
The model stays the same.
